src/api/tokensApi.js

```diff
@@ -8,28 +8,86 @@
 import requestExplorerServiceV1 from './axiosInstance';
 import { get } from 'lodash';
 
+const handleResponse = (response) => {
+  // If status is not retrieved, we assume an internal error ocurred, giving the status code 500
+  // Currently 200 is always returned for success responses
+  if (get(response, 'status', 500) !== 200) {
+    return {
+      'error': true,
+    };
+  }
+
+  return {
+    ...response,
+    error: false,
+  };
+};
+
 const tokensApi = {
-    async getList(searchText, sortBy, order, searchAfter) {
-        const data = {
-            "search_text": searchText,
-            "sort_by": sortBy,
-            "order": order,
-            "search_after": searchAfter.join(",")
-        };
-
-        let result = await requestExplorerServiceV1.get(`tokens`, { params: data });
-
-        // If status is not retrieved, we assume an internal error ocurred, giving the status code 500
-        // Currently 200 is always returned for success responses
-        if (get(result, 'status', 500) !== 200) {
-            return {
-                "error": true
-            }
-        }
-
-        result.error = false;
-        return result;
-    },
+  /**
+   * Downloads a list of tokens that match the searched text
+   *
+   * @param {String} searchText The text to match against the token values
+   * @param {String} sortBy field to sort by, possible values are (name, symbol, uid)
+   * @param {String} order Order to sort the fields, possible values are ('desc', 'asc')
+   * @param {Array[String]} searchAfter Search after this result, this should use the sort attribute of
+   * the returned hit
+   *
+   * @return {Promise}
+   */
+  async getList(searchText, sortBy, order, searchAfter) {
+    const data = {
+      'search_text': searchText,
+      'sort_by': sortBy,
+      'order': order,
+      'search_after': searchAfter.join(','),
+    };
+
+    const response = await requestExplorerServiceV1.get('tokens', { params: data });
```

Comment:
Can Axios raise some exception while running this, or does it always return something?

Reply:
I've just checked

The axios instance will never throw because we are adding an interceptor and an error handler on `axiosInstance` 

So the `handleResponse` will always receive a response (when the error handler gets an error, it will receive undefined which will default to 500)